docs/migrate-to-multi-master-guide-aws.md

```diff
@@ -0,0 +1,112 @@
+# Guide how to migrate single CP to multi cluster
+
+## before migration
+do etcd backup of etcd cluster and shut down master right after that
+```
+ETCDCTL_API=3 etcdctl snapshot save db-snapshot --key=/etc/kubernetes/ssl/etcd/client-key.pem --cert=/etc/kubernetes/ssl/etcd/client-crt.pem --cacert=/etc/kubernetes/ssl/etcd/client-ca.pem --endpoints=https://etcd.MAIN_DOMAIN:2379
+```
+dont forget to copy the etcd snapshot out of the machien to somewhere safe
+
+
+## Migration from old version
+If you are migration from older version of giantnetes-terraform which did not have multi-master support please do following steps:
+
+switch  `giantnetes-terraform` to branch `13bca31185e282a3ff65fcd523ecbef4057e2b36` and run:
+```
+terraform destroy -target=aws_cloudformation_stack.worker_asg -target=aws_instance.master --target=aws_elb.master --target=aws_elb.worker --target=aws_elb.vault --target=aws_subnet.elb_0 --target=aws_subnet.elb_1 --target=aws_subnet.worker_0 --target=aws_subnet.worker_1
+```
+
+## Apply new version
+Be sure to be on latest master branch, have prepared build dir and be inside build dir:
+```
+source envs.sh
+terraform apply ../platforms/aws/giantnetes/
+```
+
+## Migrate etcd into cluster
+
+### prepare
+* stop kubelet on all master machines
+* stop all etcd's and  delete all etcd data via `rm -rf /var/lib/etcd/member` on all masters
+```
+sudo systemctl stop etcd3 k8s-kubelet
+sudo rm -rf /var/lib/etcd/member
+```
+### master0
+Copy the etcd snapshot backup on first master and restore data from etcd backup:
+```
+ETCDCTL_API=3 etcdctl snapshot restore  --data-dir="/tmp/etcd"  db-snapshot
+sudo cp -r /tmp/etcd/member /var/lib/etcd/
+sudo rm -rf /tmp/etcd/
+sudo systemctl start etcd3
+```
+
+
+Export envs to use etcdctl againts cluster:
+```
+export ETCDCTL_CERT_FILE=/etc/kubernetes/ssl/etcd/client-crt.pem
```

Comment:
in latest master this configuration already done with alias
https://github.com/giantswarm/giantnetes-terraform/blob/master/templates/master.yaml.tmpl#L2686-L2691

Reply:
That's now is tricky. Alias references lb and endpoint will be overridden even if you export domain with `etcd1` dns

Reply:
So for migration it is make sense to add mention about cleaning up alias with `alias etcdctl=` in sudo mode

Reply:
`{{ .ETCDDomainName }}` on master is actually etcd1 or etcd2 or etcd3 depending on which master it is so we are fine

Reply:
oh, nice then